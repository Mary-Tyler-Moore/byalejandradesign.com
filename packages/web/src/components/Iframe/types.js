/** @flow */
export type Props = {
  sizes: {
    component: {
      width: number,
    },
  },
  childRef: () => HTMLElement<T>,
  ratio: number,
};
