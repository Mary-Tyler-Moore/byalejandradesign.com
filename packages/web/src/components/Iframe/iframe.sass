.iframe
  margin: 0
  width: 100%