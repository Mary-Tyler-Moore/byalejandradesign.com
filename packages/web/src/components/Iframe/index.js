/** @flow */
export { default } from './Iframe'
export * from './types'
