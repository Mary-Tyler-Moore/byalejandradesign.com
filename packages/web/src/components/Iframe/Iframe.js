/** @flow */
import * as React from 'react';
import withSize from 'react-size-components';
// styles
import './iframe.sass';

const Iframe = ({ sizes, childRef, ratio, ...props }) => (
  <iframe
    ref={childRef}
    className="iframe"
    height={sizes.component.width / ratio}
    {...props}
  />
);

Iframe.defaultProps = {
  ratio: 16 / 9,
};

export default withSize({ component: true })(Iframe);
