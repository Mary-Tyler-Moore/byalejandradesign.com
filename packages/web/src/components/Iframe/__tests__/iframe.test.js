import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Iframe from '../Iframe';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Iframe />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});

