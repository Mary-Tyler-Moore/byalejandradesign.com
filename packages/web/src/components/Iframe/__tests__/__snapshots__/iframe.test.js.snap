// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<iframe
  className="iframe"
  height={0}
/>
`;
